File: Coding/Java/Set1/SecondSmallest.java
package Set1;

public class SecondSmallest {

    public static int second_small(int[] arr){
        int smallest = Integer.MAX_VALUE;
        int second_smallest = Integer.MAX_VALUE;
        for(int num : arr){
            if(num < smallest){
                second_smallest = smallest;
                smallest = num;
            } else if(num < second_smallest && num != smallest){
                second_smallest = num;
            }
        }
        return second_smallest;
    }

    public static void main(String[] args) {
        int[] arr = {1,3,4,6,4,2};
        if(arr.length <= 1){
            System.out.println("There is no second smallest");
            System.exit(0);
        }
        int result = second_small(arr);
        if(result == Integer.MAX_VALUE){
            System.out.println("There is no second smallest");
        } else {
            System.out.println(result);
        }
    }
}

File: Coding/JavaScript/Set1/SecondSmallest.js
let arr = [1,3,4,6,4,2]
let smallest = Number.MAX_SAFE_INTEGER
let second_smallest = Number.MAX_SAFE_INTEGER
for(let val of arr){
    if(val < smallest){
        second_smallest = smallest
        smallest = val
    } else if(val < second_smallest && val != smallest){
        second_smallest = val
    }
}
if(second_smallest == Number.MAX_SAFE_INTEGER){
    console.log("there is no second smallest")
}
else
    console.log(second_smallest)